### ISSUE TITLE
Errors in the stub file

### ISSUE THREAD

**Issue by COil** [jolicode/castor#109]
Hello, I tried to include castor.php in the PHPStan analysis and there are errors because of the `.castor.stub.php` file.

<img width="877" alt="Capture d’écran 2023-06-08 à 17 10 03" src="https://github.com/jolicode/castor/assets/177844/936a77c4-cee7-496f-83d1-aeb35c95ae03">

PHPStan setup :

```
parameters:
    level: max
    paths:
        - castor.php
        - config
        - public
        - src
        - tests
    symfony:
        container_xml_path: var/cache/dev/App_KernelDevDebugContainer.xml
    # https://phpstan.org/user-guide/stub-files
    stubFiles:
        - .castor.stub.php
```

```
[16:57:54] lvernet@localdomain:/Users/coil/MicroSymfony$ castor cs:stan

stan
====

Run PHPStan
-----------

Result cache not used because the metadata do not match.
 20/20 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100% 3 secs/3 secs

Result cache is saved.
 ------ -----------------------------------------------------------------------
  Line   .castor.stub.php
 ------ -----------------------------------------------------------------------
  9      Method Castor\Attribute\AsArgument::__construct() has parameter
         $suggestedValues with no value type specified in iterable type array.
  35     Method Castor\Attribute\AsOption::__construct() has parameter
         $shortcut with no value type specified in iterable type array.
  35     Method Castor\Attribute\AsOption::__construct() has parameter
         $suggestedValues with no value type specified in iterable type array.
  44     Method Castor\Attribute\AsTask::__construct() has parameter $aliases
         with no value type specified in iterable type array.
  79     Method Castor\Console\ApplicationFactory::create() has invalid return
         type Symfony\Component\Console\SingleCommandApplication.
  79     Parameter $logger of method
         Castor\Console\ApplicationFactory::create() has invalid type
         Psr\Log\LoggerInterface.
  85     Class Castor\Console\Command\CastorFileNotFoundCommand extends
         unknown class Symfony\Component\Console\SingleCommandApplication.
  102    Property Castor\Console\Command\TaskCommand::$argumentsMap type has
         no value type specified in iterable type array.
  121    Class Castor\Context implements generic interface ArrayAccess but
         does not specify its types: TKey, TValue
  124    Method Castor\Context::__construct() has parameter $data with no
         value type specified in iterable type array.
  124    Method Castor\Context::__construct() has parameter $environment with
         no value type specified in iterable type array.
  127    Method Castor\Context::withData() has parameter $data with no value
         type specified in iterable type array.
  130    Method Castor\Context::withEnvironment() has parameter $environment
         with no value type specified in iterable type array.
  182    Property Castor\ContextRegistry::$descriptors type has no value type
         specified in iterable type array.
  196    Method Castor\ContextRegistry::getNames() return type has no value
         type specified in iterable type array.
  205    Method Castor\FunctionFinder::findFunctions() return type has no
         value type specified in iterable type iterable.
  211    Method Castor\FunctionFinder::doFindFunctions() has parameter $files
         with no value type specified in iterable type iterable.
  211    Method Castor\FunctionFinder::doFindFunctions() return type has no
         value type specified in iterable type iterable.
  223    Property Castor\GlobalHelper::$logger has unknown class
         Psr\Log\LoggerInterface as its type.
  230    Parameter $logger of method Castor\GlobalHelper::setLogger() has
         invalid type Psr\Log\LoggerInterface.
  233    Method Castor\GlobalHelper::getLogger() has invalid return type
         Psr\Log\LoggerInterface.
  252    Property Castor\SluggerHelper::$slugger has unknown class
         Symfony\Component\String\Slugger\AsciiSlugger as its type.
  259    Class Castor\Stub\NodeVisitor extends unknown class
         PhpParser\NodeVisitorAbstract.
  261    Property Castor\Stub\NodeVisitor::$currentUseStatements type has no
         value type specified in iterable type array.
  262    Method Castor\Stub\NodeVisitor::enterNode() has invalid return type
         PhpParser\Node.
  262    Parameter $node of method Castor\Stub\NodeVisitor::enterNode() has
         invalid type PhpParser\Node.
  265    Parameter $node of method Castor\Stub\NodeVisitor::leaveNode() has
         invalid type PhpParser\Node.
  322    Function Castor\parallel() return type has no value type specified in
         iterable type array.
  325    Function Castor\run() has parameter $command with no value type
         specified in iterable type array.
  325    Function Castor\run() has parameter $environment with no value type
         specified in iterable type array.
  334    Function Castor\log() has parameter $context with no value type
         specified in iterable type array.
  348    Class Symfony\Component\Console\Application implements unknown
         interface Symfony\Contracts\Service\ResetInterface.
  350    Property Symfony\Component\Console\Application::$commands type has no
         value type specified in iterable type array.
  355    Property Symfony\Component\Console\Application::$commandLoader has
         unknown class
         Symfony\Component\Console\CommandLoader\CommandLoaderInterface as its
         type.
  358    Property Symfony\Component\Console\Application::$definition has
         unknown class Symfony\Component\Console\Input\InputDefinition as its
         type.
  359    Property Symfony\Component\Console\Application::$helperSet has
         unknown class Symfony\Component\Console\Helper\HelperSet as its type.
  361    Property Symfony\Component\Console\Application::$terminal has unknown
         class Symfony\Component\Console\Terminal as its type.
  365    Property Symfony\Component\Console\Application::$signalRegistry has
         unknown class Symfony\Component\Console\SignalRegistry\SignalRegistry
         as its type.
  366    Property
         Symfony\Component\Console\Application::$signalsToDispatchEvent type
         has no value type specified in iterable type array.
  373    Method Symfony\Component\Console\Application::setCommandLoader() has
         no return type specified.
  373    Parameter $commandLoader of method
         Symfony\Component\Console\Application::setCommandLoader() has invalid
         type Symfony\Component\Console\CommandLoader\CommandLoaderInterface.
  376    Method Symfony\Component\Console\Application::getSignalRegistry() has
         invalid return type
         Symfony\Component\Console\SignalRegistry\SignalRegistry.
  379    Method
         Symfony\Component\Console\Application::setSignalsToDispatchEvent()
         has no return type specified.
  385    Method Symfony\Component\Console\Application::doRun() has no return
         type specified.
  388    Method Symfony\Component\Console\Application::reset() has no return
         type specified.
  391    Method Symfony\Component\Console\Application::setHelperSet() has no
         return type specified.
  391    Parameter $helperSet of method
         Symfony\Component\Console\Application::setHelperSet() has invalid
         type Symfony\Component\Console\Helper\HelperSet.
  394    Method Symfony\Component\Console\Application::getHelperSet() has
         invalid return type Symfony\Component\Console\Helper\HelperSet.
  397    Method Symfony\Component\Console\Application::setDefinition() has no
         return type specified.
  397    Parameter $definition of method
         Symfony\Component\Console\Application::setDefinition() has invalid
         type Symfony\Component\Console\Input\InputDefinition.
  400    Method Symfony\Component\Console\Application::getDefinition() has
         invalid return type Symfony\Component\Console\Input\InputDefinition.
  403    Parameter $input of method
         Symfony\Component\Console\Application::complete() has invalid type
         Symfony\Component\Console\Completion\CompletionInput.
  403    Parameter $suggestions of method
         Symfony\Component\Console\Application::complete() has invalid type
         Symfony\Component\Console\Completion\CompletionSuggestions.
  412    Method Symfony\Component\Console\Application::setCatchExceptions()
         has no return type specified.
  418    Method Symfony\Component\Console\Application::setAutoExit() has no
         return type specified.
  424    Method Symfony\Component\Console\Application::setName() has no return
         type specified.
  430    Method Symfony\Component\Console\Application::setVersion() has no
         return type specified.
  433    Method Symfony\Component\Console\Application::getLongVersion() has no
         return type specified.
  439    Method Symfony\Component\Console\Application::addCommands() has no
         return type specified.
  439    Method Symfony\Component\Console\Application::addCommands() has
         parameter $commands with no value type specified in iterable type
         array.
  442    Method Symfony\Component\Console\Application::add() has no return
         type specified.
  445    Method Symfony\Component\Console\Application::get() has no return
         type specified.
  451    Method Symfony\Component\Console\Application::getNamespaces() return
         type has no value type specified in iterable type array.
  457    Method Symfony\Component\Console\Application::find() has no return
         type specified.
  460    Method Symfony\Component\Console\Application::all() has no return
         type specified.
  463    Method Symfony\Component\Console\Application::getAbbreviations() has
         parameter $names with no value type specified in iterable type array.
  463    Method Symfony\Component\Console\Application::getAbbreviations()
         return type has no value type specified in iterable type array.
  472    Method Symfony\Component\Console\Application::configureIO() has no
         return type specified.
  475    Method Symfony\Component\Console\Application::doRunCommand() has no
         return type specified.
  481    Method
         Symfony\Component\Console\Application::getDefaultInputDefinition()
         has invalid return type
         Symfony\Component\Console\Input\InputDefinition.
  484    Method Symfony\Component\Console\Application::getDefaultCommands()
         return type has no value type specified in iterable type array.
  487    Method Symfony\Component\Console\Application::getDefaultHelperSet()
         has invalid return type Symfony\Component\Console\Helper\HelperSet.
  490    Method
         Symfony\Component\Console\Application::getAbbreviationSuggestions()
         has parameter $abbrevs with no value type specified in iterable type
         array.
  496    Method Symfony\Component\Console\Application::findAlternatives() has
         parameter $collection with no value type specified in iterable type
         iterable.
  496    Method Symfony\Component\Console\Application::findAlternatives()
         return type has no value type specified in iterable type array.
  505    Method Symfony\Component\Console\Application::splitStringByWidth()
         return type has no value type specified in iterable type array.
  508    Method Symfony\Component\Console\Application::extractAllNamespaces()
         return type has no value type specified in iterable type array.
  524    Property Symfony\Component\Console\Input\InputArgument::$default type
         has no value type specified in iterable type array.
  525    Property
         Symfony\Component\Console\Input\InputArgument::$suggestedValues type
         has no value type specified in iterable type array.
  527    Method Symfony\Component\Console\Input\InputArgument::__construct()
         has parameter $default with no value type specified in iterable type
         array.
  527    Method Symfony\Component\Console\Input\InputArgument::__construct()
         has parameter $suggestedValues with no value type specified in
         iterable type array.
  539    Method Symfony\Component\Console\Input\InputArgument::setDefault()
         has no return type specified.
  539    Method Symfony\Component\Console\Input\InputArgument::setDefault()
         has parameter $default with no value type specified in iterable type
         array.
  542    Method Symfony\Component\Console\Input\InputArgument::getDefault()
         return type has no value type specified in iterable type array.
  548    Parameter $input of method
         Symfony\Component\Console\Input\InputArgument::complete() has invalid
         type Symfony\Component\Console\Completion\CompletionInput.
  548    Parameter $suggestions of method
         Symfony\Component\Console\Input\InputArgument::complete() has invalid
         type Symfony\Component\Console\Completion\CompletionSuggestions.
  562    Method
         Symfony\Component\Console\Input\InputInterface::hasParameterOption()
         has parameter $values with no value type specified in iterable type
         array.
  565    Method
         Symfony\Component\Console\Input\InputInterface::getParameterOption()
         has no return type specified.
  565    Method
         Symfony\Component\Console\Input\InputInterface::getParameterOption()
         has parameter $default with no value type specified in iterable type
         array.
  565    Method
         Symfony\Component\Console\Input\InputInterface::getParameterOption()
         has parameter $values with no value type specified in iterable type
         array.
  568    Method Symfony\Component\Console\Input\InputInterface::bind() has no
         return type specified.
  568    Parameter $definition of method
         Symfony\Component\Console\Input\InputInterface::bind() has invalid
         type Symfony\Component\Console\Input\InputDefinition.
  571    Method Symfony\Component\Console\Input\InputInterface::validate() has
         no return type specified.
  574    Method Symfony\Component\Console\Input\InputInterface::getArguments()
         return type has no value type specified in iterable type array.
  577    Method Symfony\Component\Console\Input\InputInterface::getArgument()
         has no return type specified.
  580    Method Symfony\Component\Console\Input\InputInterface::setArgument()
         has no return type specified.
  586    Method Symfony\Component\Console\Input\InputInterface::getOptions()
         return type has no value type specified in iterable type array.
  589    Method Symfony\Component\Console\Input\InputInterface::getOption()
         has no return type specified.
  592    Method Symfony\Component\Console\Input\InputInterface::setOption()
         has no return type specified.
  601    Method
         Symfony\Component\Console\Input\InputInterface::setInteractive() has
         no return type specified.
  615    Property Symfony\Component\Console\Input\InputOption::$shortcut type
         has no value type specified in iterable type array.
  617    Property Symfony\Component\Console\Input\InputOption::$default type
         has no value type specified in iterable type array.
  618    Property
         Symfony\Component\Console\Input\InputOption::$suggestedValues type
         has no value type specified in iterable type array.
  620    Method Symfony\Component\Console\Input\InputOption::__construct() has
         parameter $default with no value type specified in iterable type
         array.
  620    Method Symfony\Component\Console\Input\InputOption::__construct() has
         parameter $shortcut with no value type specified in iterable type
         array.
  620    Method Symfony\Component\Console\Input\InputOption::__construct() has
         parameter $suggestedValues with no value type specified in iterable
         type array.
  644    Method Symfony\Component\Console\Input\InputOption::setDefault() has
         no return type specified.
  644    Method Symfony\Component\Console\Input\InputOption::setDefault() has
         parameter $default with no value type specified in iterable type
         array.
  647    Method Symfony\Component\Console\Input\InputOption::getDefault()
         return type has no value type specified in iterable type array.
  656    Parameter $input of method
         Symfony\Component\Console\Input\InputOption::complete() has invalid
         type Symfony\Component\Console\Completion\CompletionInput.
  656    Parameter $suggestions of method
         Symfony\Component\Console\Input\InputOption::complete() has invalid
         type Symfony\Component\Console\Completion\CompletionSuggestions.
  675    Method Symfony\Component\Console\Output\OutputInterface::write() has
         no return type specified.
  675    Method Symfony\Component\Console\Output\OutputInterface::write() has
         parameter $messages with no value type specified in iterable type
         iterable.
  678    Method Symfony\Component\Console\Output\OutputInterface::writeln()
         has no return type specified.
  678    Method Symfony\Component\Console\Output\OutputInterface::writeln()
         has parameter $messages with no value type specified in iterable type
         iterable.
  681    Method
         Symfony\Component\Console\Output\OutputInterface::setVerbosity() has
         no return type specified.
  699    Method
         Symfony\Component\Console\Output\OutputInterface::setDecorated() has
         no return type specified.
  705    Method
         Symfony\Component\Console\Output\OutputInterface::setFormatter() has
         no return type specified.
  705    Parameter $formatter of method
         Symfony\Component\Console\Output\OutputInterface::setFormatter() has
         invalid type
         Symfony\Component\Console\Formatter\OutputFormatterInterface.
  708    Method
         Symfony\Component\Console\Output\OutputInterface::getFormatter() has
         invalid return type
         Symfony\Component\Console\Formatter\OutputFormatterInterface.
  714    Class Symfony\Component\Console\Style\SymfonyStyle extends unknown
         class Symfony\Component\Console\Style\OutputStyle.
  719    Property
         Symfony\Component\Console\Style\SymfonyStyle::$questionHelper has
         unknown class Symfony\Component\Console\Helper\SymfonyQuestionHelper
         as its type.
  720    Property Symfony\Component\Console\Style\SymfonyStyle::$progressBar
         has unknown class Symfony\Component\Console\Helper\ProgressBar as its
         type.
  722    Property
         Symfony\Component\Console\Style\SymfonyStyle::$bufferedOutput has
         unknown class Symfony\Component\Console\Output\TrimmedBufferOutput as
         its type.
  726    Method Symfony\Component\Console\Style\SymfonyStyle::block() has no
         return type specified.
  726    Method Symfony\Component\Console\Style\SymfonyStyle::block() has
         parameter $messages with no value type specified in iterable type
         array.
  729    Method Symfony\Component\Console\Style\SymfonyStyle::title() has no
         return type specified.
  732    Method Symfony\Component\Console\Style\SymfonyStyle::section() has no
         return type specified.
  735    Method Symfony\Component\Console\Style\SymfonyStyle::listing() has no
         return type specified.
  735    Method Symfony\Component\Console\Style\SymfonyStyle::listing() has
         parameter $elements with no value type specified in iterable type
         array.
  738    Method Symfony\Component\Console\Style\SymfonyStyle::text() has no
         return type specified.
  738    Method Symfony\Component\Console\Style\SymfonyStyle::text() has
         parameter $message with no value type specified in iterable type
         array.
  741    Method Symfony\Component\Console\Style\SymfonyStyle::comment() has no
         return type specified.
  741    Method Symfony\Component\Console\Style\SymfonyStyle::comment() has
         parameter $message with no value type specified in iterable type
         array.
  744    Method Symfony\Component\Console\Style\SymfonyStyle::success() has no
         return type specified.
  744    Method Symfony\Component\Console\Style\SymfonyStyle::success() has
         parameter $message with no value type specified in iterable type
         array.
  747    Method Symfony\Component\Console\Style\SymfonyStyle::error() has no
         return type specified.
  747    Method Symfony\Component\Console\Style\SymfonyStyle::error() has
         parameter $message with no value type specified in iterable type
         array.
  750    Method Symfony\Component\Console\Style\SymfonyStyle::warning() has no
         return type specified.
  750    Method Symfony\Component\Console\Style\SymfonyStyle::warning() has
         parameter $message with no value type specified in iterable type
         array.
  753    Method Symfony\Component\Console\Style\SymfonyStyle::note() has no
         return type specified.
  753    Method Symfony\Component\Console\Style\SymfonyStyle::note() has
         parameter $message with no value type specified in iterable type
         array.
  756    Method Symfony\Component\Console\Style\SymfonyStyle::info() has no
         return type specified.
  756    Method Symfony\Component\Console\Style\SymfonyStyle::info() has
         parameter $message with no value type specified in iterable type
         array.
  759    Method Symfony\Component\Console\Style\SymfonyStyle::caution() has no
         return type specified.
  759    Method Symfony\Component\Console\Style\SymfonyStyle::caution() has
         parameter $message with no value type specified in iterable type
         array.
  762    Method Symfony\Component\Console\Style\SymfonyStyle::table() has no
         return type specified.
  762    Method Symfony\Component\Console\Style\SymfonyStyle::table() has
         parameter $headers with no value type specified in iterable type
         array.
  762    Method Symfony\Component\Console\Style\SymfonyStyle::table() has
         parameter $rows with no value type specified in iterable type array.
  765    Method
         Symfony\Component\Console\Style\SymfonyStyle::horizontalTable() has
         no return type specified.
  765    Method
         Symfony\Component\Console\Style\SymfonyStyle::horizontalTable() has
         parameter $headers with no value type specified in iterable type
         array.
  765    Method
         Symfony\Component\Console\Style\SymfonyStyle::horizontalTable() has
         parameter $rows with no value type specified in iterable type array.
  768    Method Symfony\Component\Console\Style\SymfonyStyle::definitionList()
         has no return type specified.
  768    Method Symfony\Component\Console\Style\SymfonyStyle::definitionList()
         has parameter $list with no value type specified in iterable type
         array.
  768    Parameter $list of method
         Symfony\Component\Console\Style\SymfonyStyle::definitionList() has
         invalid type Symfony\Component\Console\Helper\TableSeparator.
  780    Method Symfony\Component\Console\Style\SymfonyStyle::choice() has
         parameter $choices with no value type specified in iterable type
         array.
  783    Method Symfony\Component\Console\Style\SymfonyStyle::progressStart()
         has no return type specified.
  786    Method
         Symfony\Component\Console\Style\SymfonyStyle::progressAdvance() has
         no return type specified.
  789    Method Symfony\Component\Console\Style\SymfonyStyle::progressFinish()
         has no return type specified.
  792    Method
         Symfony\Component\Console\Style\SymfonyStyle::createProgressBar() has
         invalid return type Symfony\Component\Console\Helper\ProgressBar.
  795    Method
         Symfony\Component\Console\Style\SymfonyStyle::progressIterate() has
         parameter $iterable with no value type specified in iterable type
         iterable.
  795    Method
         Symfony\Component\Console\Style\SymfonyStyle::progressIterate()
         return type has no value type specified in iterable type iterable.
  798    Parameter $question of method
         Symfony\Component\Console\Style\SymfonyStyle::askQuestion() has
         invalid type Symfony\Component\Console\Question\Question.
  801    Method Symfony\Component\Console\Style\SymfonyStyle::writeln() has no
         return type specified.
  801    Method Symfony\Component\Console\Style\SymfonyStyle::writeln() has
         parameter $messages with no value type specified in iterable type
         iterable.
  804    Method Symfony\Component\Console\Style\SymfonyStyle::write() has no
         return type specified.
  804    Method Symfony\Component\Console\Style\SymfonyStyle::write() has
         parameter $messages with no value type specified in iterable type
         iterable.
  807    Method Symfony\Component\Console\Style\SymfonyStyle::newLine() has no
         return type specified.
  813    Method Symfony\Component\Console\Style\SymfonyStyle::createTable()
         has invalid return type Symfony\Component\Console\Helper\Table.
  816    Method Symfony\Component\Console\Style\SymfonyStyle::getProgressBar()
         has invalid return type Symfony\Component\Console\Helper\ProgressBar.
  828    Method Symfony\Component\Console\Style\SymfonyStyle::createBlock()
         has parameter $messages with no value type specified in iterable type
         iterable.
  828    Method Symfony\Component\Console\Style\SymfonyStyle::createBlock()
         return type has no value type specified in iterable type array.
  836    Property Symfony\Component\Filesystem\Filesystem::$lastError has no
         type specified.
  837    Method Symfony\Component\Filesystem\Filesystem::copy() has no return
         type specified.
  840    Method Symfony\Component\Filesystem\Filesystem::mkdir() has no return
         type specified.
  840    Method Symfony\Component\Filesystem\Filesystem::mkdir() has parameter
         $dirs with no value type specified in iterable type iterable.
  843    Method Symfony\Component\Filesystem\Filesystem::exists() has
         parameter $files with no value type specified in iterable type
         iterable.
  846    Method Symfony\Component\Filesystem\Filesystem::touch() has no return
         type specified.
  846    Method Symfony\Component\Filesystem\Filesystem::touch() has parameter
         $files with no value type specified in iterable type iterable.
  849    Method Symfony\Component\Filesystem\Filesystem::remove() has no
         return type specified.
  849    Method Symfony\Component\Filesystem\Filesystem::remove() has
         parameter $files with no value type specified in iterable type
         iterable.
  852    Method Symfony\Component\Filesystem\Filesystem::doRemove() has
         parameter $files with no value type specified in iterable type array.
  855    Method Symfony\Component\Filesystem\Filesystem::chmod() has no return
         type specified.
  855    Method Symfony\Component\Filesystem\Filesystem::chmod() has parameter
         $files with no value type specified in iterable type iterable.
  858    Method Symfony\Component\Filesystem\Filesystem::chown() has no return
         type specified.
  858    Method Symfony\Component\Filesystem\Filesystem::chown() has parameter
         $files with no value type specified in iterable type iterable.
  861    Method Symfony\Component\Filesystem\Filesystem::chgrp() has no return
         type specified.
  861    Method Symfony\Component\Filesystem\Filesystem::chgrp() has parameter
         $files with no value type specified in iterable type iterable.
  864    Method Symfony\Component\Filesystem\Filesystem::rename() has no
         return type specified.
  870    Method Symfony\Component\Filesystem\Filesystem::symlink() has no
         return type specified.
  873    Method Symfony\Component\Filesystem\Filesystem::hardlink() has no
         return type specified.
  873    Method Symfony\Component\Filesystem\Filesystem::hardlink() has
         parameter $targetFiles with no value type specified in iterable type
         iterable.
  885    Method Symfony\Component\Filesystem\Filesystem::mirror() has no
         return type specified.
  885    Method Symfony\Component\Filesystem\Filesystem::mirror() has
         parameter $iterator with no value type specified in iterable type
         Traversable.
  885    Method Symfony\Component\Filesystem\Filesystem::mirror() has
         parameter $options with no value type specified in iterable type
         array.
  894    Method Symfony\Component\Filesystem\Filesystem::dumpFile() has no
         return type specified.
  894    Method Symfony\Component\Filesystem\Filesystem::dumpFile() has
         parameter $content with no type specified.
  897    Method Symfony\Component\Filesystem\Filesystem::appendToFile() has no
         return type specified.
  897    Method Symfony\Component\Filesystem\Filesystem::appendToFile() has
         parameter $content with no type specified.
  900    Method Symfony\Component\Filesystem\Filesystem::toIterable() has
         parameter $files with no value type specified in iterable type
         iterable.
  900    Method Symfony\Component\Filesystem\Filesystem::toIterable() return
         type has no value type specified in iterable type iterable.
  903    Method
         Symfony\Component\Filesystem\Filesystem::getSchemeAndHierarchy()
         return type has no value type specified in iterable type array.
  922    Property Symfony\Component\Filesystem\Path::$buffer has no type
         specified.
  923    Property Symfony\Component\Filesystem\Path::$bufferSize has no type
         specified.
  945    Method Symfony\Component\Filesystem\Path::hasExtension() has
         parameter $extensions with no type specified.
  975    Method Symfony\Component\Filesystem\Path::findCanonicalParts() return
         type has no value type specified in iterable type array.
  978    Method Symfony\Component\Filesystem\Path::split() return type has no
         value type specified in iterable type array.
  993    Class Symfony\Component\Finder\Finder implements generic interface
         IteratorAggregate but does not specify its types: TKey, TValue
  993    PHPDoc tag @implements has invalid value (): Unexpected token "\n",
         expected type at offset 15
  999    Property Symfony\Component\Finder\Finder::$names type has no value
         type specified in iterable type array.
  1000   Property Symfony\Component\Finder\Finder::$notNames type has no value
         type specified in iterable type array.
  1001   Property Symfony\Component\Finder\Finder::$exclude type has no value
         type specified in iterable type array.
  1002   Property Symfony\Component\Finder\Finder::$filters type has no value
         type specified in iterable type array.
  1003   Property Symfony\Component\Finder\Finder::$depths type has no value
         type specified in iterable type array.
  1004   Property Symfony\Component\Finder\Finder::$sizes type has no value
         type specified in iterable type array.
  1009   Property Symfony\Component\Finder\Finder::$dirs type has no value
         type specified in iterable type array.
  1010   Property Symfony\Component\Finder\Finder::$dates type has no value
         type specified in iterable type array.
  1011   Property Symfony\Component\Finder\Finder::$iterators type has no
         value type specified in iterable type array.
  1012   Property Symfony\Component\Finder\Finder::$contains type has no value
         type specified in iterable type array.
  1013   Property Symfony\Component\Finder\Finder::$notContains type has no
         value type specified in iterable type array.
  1014   Property Symfony\Component\Finder\Finder::$paths type has no value
         type specified in iterable type array.
  1015   Property Symfony\Component\Finder\Finder::$notPaths type has no value
         type specified in iterable type array.
  1017   Property Symfony\Component\Finder\Finder::$vcsPatterns type has no
         value type specified in iterable type array.
  1030   Method Symfony\Component\Finder\Finder::depth() has parameter $levels
         with no value type specified in iterable type array.
  1033   Method Symfony\Component\Finder\Finder::date() has parameter $dates
         with no value type specified in iterable type array.
  1036   Method Symfony\Component\Finder\Finder::name() has parameter
         $patterns with no value type specified in iterable type array.
  1039   Method Symfony\Component\Finder\Finder::notName() has parameter
         $patterns with no value type specified in iterable type array.
  1042   Method Symfony\Component\Finder\Finder::contains() has parameter
         $patterns with no value type specified in iterable type array.
  1045   Method Symfony\Component\Finder\Finder::notContains() has parameter
         $patterns with no value type specified in iterable type array.
  1048   Method Symfony\Component\Finder\Finder::path() has parameter
         $patterns with no value type specified in iterable type array.
  1051   Method Symfony\Component\Finder\Finder::notPath() has parameter
         $patterns with no value type specified in iterable type array.
  1054   Method Symfony\Component\Finder\Finder::size() has parameter $sizes
         with no value type specified in iterable type array.
  1057   Method Symfony\Component\Finder\Finder::exclude() has parameter $dirs
         with no value type specified in iterable type array.
  1069   Method Symfony\Component\Finder\Finder::addVCSPattern() has no return
         type specified.
  1069   Method Symfony\Component\Finder\Finder::addVCSPattern() has parameter
         $pattern with no value type specified in iterable type array.
  1111   Method Symfony\Component\Finder\Finder::in() has parameter $dirs with
         no value type specified in iterable type array.
  1117   Method Symfony\Component\Finder\Finder::append() has parameter
         $iterator with no value type specified in iterable type iterable.
  1138   Class Symfony\Component\Process\Process implements generic interface
         IteratorAggregate but does not specify its types: TKey, TValue
  1138   PHPDoc tag @implements has invalid value (): Unexpected token "\n",
         expected type at offset 15
  1153   Property Symfony\Component\Process\Process::$callback has no type
         specified.
  1154   Property Symfony\Component\Process\Process::$hasCallback has no type
         specified.
  1155   Property Symfony\Component\Process\Process::$commandline has no type
         specified.
  1156   Property Symfony\Component\Process\Process::$cwd has no type
         specified.
  1157   Property Symfony\Component\Process\Process::$env has no type
         specified.
  1158   Property Symfony\Component\Process\Process::$input has no type
         specified.
  1159   Property Symfony\Component\Process\Process::$starttime has no type
         specified.
  1160   Property Symfony\Component\Process\Process::$lastOutputTime has no
         type specified.
  1161   Property Symfony\Component\Process\Process::$timeout has no type
         specified.
  1162   Property Symfony\Component\Process\Process::$idleTimeout has no type
         specified.
  1163   Property Symfony\Component\Process\Process::$exitcode has no type
         specified.
  1164   Property Symfony\Component\Process\Process::$fallbackStatus has no
         type specified.
  1165   Property Symfony\Component\Process\Process::$processInformation has
         no type specified.
  1166   Property Symfony\Component\Process\Process::$outputDisabled has no
         type specified.
  1167   Property Symfony\Component\Process\Process::$stdout has no type
         specified.
  1168   Property Symfony\Component\Process\Process::$stderr has no type
         specified.
  1169   Property Symfony\Component\Process\Process::$process has no type
         specified.
  1170   Property Symfony\Component\Process\Process::$status has no type
         specified.
  1171   Property Symfony\Component\Process\Process::$incrementalOutputOffset
         has no type specified.
  1172   Property
         Symfony\Component\Process\Process::$incrementalErrorOutputOffset has
         no type specified.
  1173   Property Symfony\Component\Process\Process::$tty has no type
         specified.
  1174   Property Symfony\Component\Process\Process::$pty has no type
         specified.
  1175   Property Symfony\Component\Process\Process::$options has no type
         specified.
  1176   Property Symfony\Component\Process\Process::$useFileHandles has no
         type specified.
  1177   Property Symfony\Component\Process\Process::$processPipes has no type
         specified.
  1178   Property Symfony\Component\Process\Process::$latestSignal has no type
         specified.
  1179   Property Symfony\Component\Process\Process::$sigchild has no type
         specified.
  1180   Property Symfony\Component\Process\Process::$exitCodes has no type
         specified.
  1181   Method Symfony\Component\Process\Process::__construct() has parameter
         $command with no value type specified in iterable type array.
  1181   Method Symfony\Component\Process\Process::__construct() has parameter
         $env with no value type specified in iterable type array.
  1184   Method Symfony\Component\Process\Process::fromShellCommandline() has
         parameter $env with no value type specified in iterable type array.
  1199   Method Symfony\Component\Process\Process::run() has parameter $env
         with no value type specified in iterable type array.
  1202   Method Symfony\Component\Process\Process::mustRun() has parameter
         $env with no value type specified in iterable type array.
  1205   Method Symfony\Component\Process\Process::start() has no return type
         specified.
  1205   Method Symfony\Component\Process\Process::start() has parameter $env
         with no value type specified in iterable type array.
  1208   Method Symfony\Component\Process\Process::restart() has parameter
         $env with no value type specified in iterable type array.
  1331   Method Symfony\Component\Process\Process::getEnv() return type has no
         value type specified in iterable type array.
  1334   Method Symfony\Component\Process\Process::setEnv() has parameter $env
         with no value type specified in iterable type array.
  1337   Method Symfony\Component\Process\Process::getInput() has no return
         type specified.
  1343   Method Symfony\Component\Process\Process::checkTimeout() has no
         return type specified.
  1349   Method Symfony\Component\Process\Process::setOptions() has no return
         type specified.
  1349   Method Symfony\Component\Process\Process::setOptions() has parameter
         $options with no value type specified in iterable type array.
  1358   Method Symfony\Component\Process\Process::getDescriptors() return
         type has no value type specified in iterable type array.
  1364   Method Symfony\Component\Process\Process::updateStatus() has no
         return type specified.
  1388   Method Symfony\Component\Process\Process::prepareWindowsCommandLine()
         has parameter $env with no value type specified in iterable type
         array.
  1400   Method Symfony\Component\Process\Process::replacePlaceholders() has
         parameter $env with no value type specified in iterable type array.
  1403   Method Symfony\Component\Process\Process::getDefaultEnv() return type
         has no value type specified in iterable type array.
 ------ -----------------------------------------------------------------------

 ------ ---------------------------------------------------------------------
  Line   castor.php
 ------ ---------------------------------------------------------------------
  9      Used function Castor\run not found.
         💡 Learn more at https://phpstan.org/user-guide/discovering-symbols
  21     Attribute class Castor\Attribute\AsTask does not exist.
```

**Comment by lyrixx** [comment]
There were errors in the stub files. They have been fixed in #116.
Thanks for the report!

However, I see PHPStan returns errors from the stubs itself:

1. The first error reported:
    > 9      Method Castor\Attribute\AsArgument::__construct() has parameter
    >         $suggestedValues with no value type specified in iterable type array
    But I can see this in the stub:
    ```php
    #[\Attribute(\Attribute::TARGET_PARAMETER)]
    class AsArgument extends AsCommandArgument
    {
        /**
         * @param array<string> $suggestedValues
         */
        public function __construct(string|null $name = null, public readonly string $description = '', public readonly array $suggestedValues = [])
        {
        }
    }
    ```
    So I guess PHPStan is wrong, doesn't it?

2. There are some report for our vendor, like Symfony, and we cannot do anything for this.


That's why it does make sens to me to analyse such files. It could be used to enhance covering of other task, it must not be analyzed (or add it to your baseline)

**Comment by lyrixx** [comment]
For the record, you must not used `stubFiles`, but `scanFiles`

see 
* https://phpstan.org/user-guide/stub-files
* https://phpstan.org/user-guide/discovering-symbols#third-party-code-outside-of-composer-dependencies


TL;DR: stub is not for stubs (lol) but for PHP Doc

**Comment by COil** [comment]
Oh thanks, I'll give a try.
